### DISCUSSION TITLE
Option to apply "Ignore when blank" to all fields in the WorkBench

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5705] [Feature Requests]
It would be helpful to have a toggle to apply the "ignore when blank" setting to all fields

> **Example:** making a template for collection object upload that includes columns for attributes/determinations/preparations - sometimes there won't be any related objects so we don't want to accidentally create an empty row in the linked tables

I replied:

> Regarding your first point, if you do not have any values in a table's columns, it should not create a blank record. 
> 
> See this screenshot where I had values in Collection Objet and Accession but nothing in the Agent relationship. No record was created in the empty table. The ignore when blank is generally useful for associating a record with an existing one (e.g. You have an Agent where you know the first and last name but the middle name was not recorded for all records. You want to make sure that it matches an existing agent even if the middle initial column is empty and the existing agent has a middle name in the database)
>
> ![image](https://user-images.githubusercontent.com/37256050/210569602-956cfc53-6bf9-40e1-8a2b-1786ae1ee99c.png)

**Requested By:** Dan Baker at CSIRO ([Asana Link](https://app.asana.com/0/1201972579963460/1203637920507460/f))

**Comment by danb213** [comment]
Thanks for the clarification on the issue for having no object created if all table columns are blank, but it would still be a good feature to have especially for your agent example - we will almost always want to turn that option on for all agent columns so a "toggle all" button would be nice.

**Comment by maxpatiiuk** [comment]
Should do this as either a user pref or as an option in the wb plan to bulk set matching option for all fields
